Gems/Atom/RPI/Code/Source/RPI.Builders/Material/MaterialTypeBuilder.cpp

```diff
@@ -327,7 +327,32 @@ namespace AZ
 
             if (auto settingsRegistry = AZ::SettingsRegistry::Get(); settingsRegistry != nullptr)
             {
-                settingsRegistry->GetObject(materialPipelines, "/O3DE/Atom/RPI/MaterialPipelineFiles");
+                const char* materialPipelineJsonPath = "/O3DE/Atom/RPI/MaterialPipelineFiles";
```

Comment:
Suggestion: Rather than changing the format of the original container and warning the user, you could leave the original container in place and add a secondary container. 

Then you would have something like
"/O3DE/Atom/RPI/MaterialPipelineFiles" (the original vector)
"/O3DE/Atom/RPI/MaterialPipelineFilesByGem" (additional material pipelines per gem)
You could read the original first and then append all of the pipelines from the gem maps to the same container. Then developers are still free to add different material pipelines per project without having a gem name. By supporting both, you preserve backward compatibility and don't need to check the type beforehand.

Reply:
Changed it to use two containers as suggested